incubator/hnc/internal/reconcilers/object.go

```diff
@@ -787,14 +790,29 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getSelector returns the selector on a given object if it exists
+func (r *ObjectReconciler) getTreeSelector(log logr.Logger, inst *unstructured.Unstructured) labels.Selector {
+	annot := inst.GetAnnotations()
+	treeSelectorStr, ok := annot[api.AnnotationTreeSelector]
+	if !ok {
+		return nil
+	}
+	return r.getSelectorFromString(log, treeSelectorStr+api.LabelTreeDepthSuffix)
```

Comment:
This doesn't look right to me - we specified that there could be multiple comma-separated trees in the tree-selector, but this doesn't account for that, does it?

This probably shows that we need tests (both tree and regular) showing more complex selectors. Can you please add a more complex test for the tree selectors in this PR?

Reply:
done